File: ketch-android/webview/Android Ketch Pref Center using Java/app/src/main/java/com/ketch/sample/webview/KetchSharedPreferences.java
package com.ketch.sample.webview;

import android.content.Context;
import android.content.SharedPreferences;

public class KetchSharedPreferences {
    private static final String IAB_TCF_TC_STRING = "IABTCF_TCString";
    private static final String IAB_US_PRIVACY_STRING = "IABUSPrivacy_String";

    private final SharedPreferences sharedPreferences;

    public KetchSharedPreferences(Context context) {
        sharedPreferences = context.getSharedPreferences(context.getPackageName(), Context.MODE_PRIVATE);
    }

    public String getUSPrivacyString() {
        return sharedPreferences.getString(IAB_US_PRIVACY_STRING, null);
    }

    public void saveUSPrivacyString(String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        if (value != null) {
            editor.putString(IAB_US_PRIVACY_STRING, value);
        } else {
            editor.remove(IAB_US_PRIVACY_STRING);
        }
        editor.apply();
    }

    public String getTCFTCString() {
        return sharedPreferences.getString(IAB_TCF_TC_STRING, null);
    }

    public void saveTCFTCString(String value) {
        SharedPreferences.Editor editor = sharedPreferences.edit();
        if (value != null) {
            editor.putString(IAB_TCF_TC_STRING, value);
        } else {
            editor.remove(IAB_TCF_TC_STRING);
        }
        editor.apply();
    }
}

File: ketch-android/Android Ketch Pref Center using Kotlin/app/src/main/java/com/ketch/sample/pref/KetchSharedPreferences.kt
package com.ketch.sample.pref

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit


class KetchSharedPreferences(context: Context) {
    private val sharedPreferences: SharedPreferences

    init {
        sharedPreferences = context.getSharedPreferences(context.packageName, Context.MODE_PRIVATE)
    }

    fun getUSPrivacyString(): String? = sharedPreferences.getString(IAB_US_PRIVACY_STRING, null)

    fun saveUSPrivacyString(value: String?) {
        sharedPreferences.edit {
            value?.let {
                putString(IAB_US_PRIVACY_STRING, value)
            } ?: remove(IAB_US_PRIVACY_STRING)
            apply()
        }
    }

    fun getTCFTCString(): String? = sharedPreferences.getString(IAB_TCF_TC_STRING, null)

    fun saveTCFTCString(value: String?) {
        sharedPreferences.edit {
            value?.let {
                putString(IAB_TCF_TC_STRING, value)
            } ?: remove(IAB_TCF_TC_STRING)
            apply()
        }
    }

    companion object {
        private const val IAB_TCF_TC_STRING = "IABTCF_TCString"
        private const val IAB_US_PRIVACY_STRING = "IABUSPrivacy_String"
    }
}
